Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,559 @@
+  if not UI.IsMovementPathOn() or UI.IsGameCoreBusy() then
+    return;
+  end
+    
+  -- CQUI (Azurency) : Check if in CITY_MANAGEMENT or STRIKE modes
+  local CQUI_im = UI.GetInterfaceMode();
+  if (CQUI_im == InterfaceModeTypes.CITY_MANAGEMENT or CQUI_im == InterfaceModeTypes.CITY_RANGE_ATTACK or CQUI_im == InterfaceModeTypes.DISTRICT_RANGE_ATTACK) then
+    return;
+  end
+
+  return BASE_CQUI_RealizeMovementPath( showQueuedPath );
+end
+
+-- ===========================================================================
+function OnUnitSelectionChanged( playerID:number, unitID:number, hexI:number, hexJ:number, hexK:number, isSelected:boolean, isEditable:boolean )
+  local msg = "** Function Entry: OnUnitSelectionChanged (CQUI Hook).  playerId:"..tostring(playerID).." unitId:"..tostring(unitID).." hexI:"..tostring(hexI).." hexJ:"..tostring(hexJ).." hexK:"..tostring(hexK).." isSelected:"..tostring(isSelected).." isEditable:"..tostring(isEditable);
+  print_debug(msg);
+  if playerID ~= Game.GetLocalPlayer() then
+    return;
+  end
+
+  -- CQUI (Azurency) : Fixes a Vanilla bug from SelectUnit.lua (not taking in account the district range attack)
+  -- CQUI (Azurency) : If a selection is occuring and the district attack interface mode is up, take it down.
+  if UI.GetInterfaceMode() == InterfaceModeTypes.DISTRICT_RANGE_ATTACK then
+    UI.SetInterfaceMode(InterfaceModeTypes.SELECTION);
+  end
+
+  return BASE_CQUI_OnUnitSelectionChanged( playerID, unitID, hexI, hexJ, hexK, isSelected, isEditable );
+end
+
+-- ===========================================================================
+function DefaultKeyDownHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyDownHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+  -- Note: This function always returns false, by design in Base game
+  BASE_CQUI_DefaultKeyDownHandler( uiKey ) ;
+
+  --CQUI Keybinds
+  local keyPanChanged :boolean = false;
+
+  if uiKey == Keys.VK_SHIFT then
+    print_debug("CQUI_isShiftDown = true");
+    CQUI_isShiftDown = true;
+  end
+
+  if uiKey == Keys.VK_ALT then
+    print_debug("CQUI_isAltDown = true");
+    CQUI_isAltDown = true;
+  end
+
+  if CQUI_hotkeyMode ~= CQUI_HOTKEYMODE_STANDARD then
+    if CQUI_hotkeyMode == CQUI_HOTKEYMODE_ENHANCED then
+      if (uiKey == Keys.W) then
+        keyPanChanged = true;
+        m_isUPpressed = true;
+      end
+
+      if (uiKey == Keys.D) then
+        keyPanChanged = true;
+        m_isRIGHTpressed = true;
+      end
+
+      if (uiKey == Keys.S) then
+        keyPanChanged = true;
+        m_isDOWNpressed = true;
+      end
+
+      if (uiKey == Keys.A) then
+        keyPanChanged = true;
+        m_isLEFTpressed = true;
+      end
+    end
+  end
+
+  if (keyPanChanged == true) then
+    -- Base game file uses m_edgePanX and m_edgePanY... but those values do not appear to ever change from 0, so just send 0.
+    ProcessPan(0, 0);
+    return true;
```

Comment:
Line above seems new. Any reason to add it?

Reply:
it's relocated from the original end of that function.  The original end of the function also had handlers for the arrow keys, before calling this method.  I removed the arrow key handlers (just let the base game handle them), it didn't make sense to have the ProcessPan call be at the bottom of that function anymore, so i moved it closer to the code elements that use it

Reply:
I was referring to `return true;`. Doesn't the original always `return false;`?

Reply:
I think this is answered by your answer to my other my comment regarding key stroke being consumed.